sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/pushrefresh/AppConfigurationRefreshEndpoint.java

```diff
@@ -81,15 +85,24 @@ public String refresh(HttpServletRequest request, HttpServletResponse response,
             return HttpStatus.UNAUTHORIZED.getReasonPhrase();
         }
 
+        String syncToken = "";
+
+        JsonNode data = kvReference.findValue(DATA);
+        if (data != null) {
+            JsonNode syncTokenNode = data.findValue(SYNC_TOKEN);
+            if (syncTokenNode != null) {
+                syncToken = syncTokenNode.asText();
+            }
+        }
+
```

Comment:
This seems like duplicate code with the one in `AppConfigurationBusRefreshEndpoint`, maybe we could create a util class and extract it into a method?

Reply:
I reworked this.